src/parseConfigFile.ts

```diff
@@ -0,0 +1,51 @@
+import * as fs from "fs";
+import { Cell, CellValue, GameState } from "./gamestate";
+
+export function loadGameStateFromFile(filePath: string): GameState {
+  let data = parseConfigFile(filePath);
+  if (!Array.isArray(data)) {
+    throw Error(`Malformed config file ${filePath}`);
+  }
+  let rowsCount = data.length;
+  let gameState = { board: [] };
+  for (let row of data) {
+    gameState.board.push(parseRow(row, rowsCount, filePath));
+  }
+  return gameState;
+}
+
+function parseConfigFile(filePath: string): any {
+  try {
+    return JSON.parse(fs.readFileSync(filePath, "utf-8"));
+  } catch (error) {
+    console.error(error);
+    throw Error(`Unable to parse config file ${filePath}`);
+  }
+}
+
+function parseRow(row: any, rowsCount: number, filePath: string): Array<Cell> {
```

Comment:
row should be typed as you iterate over it in the function

Reply:
Indeed I could have used `JSON.parse(fileContent) as Array<Array<Cell>>` or stg close, but we still need to map each cell to a `Cell` object, which would have to be done manually.

Keeping actual impl for now